src/zc/sourcefactory/browser/mapping.py

```diff
@@ -24,12 +24,12 @@
 
 @zope.interface.implementer(zope.browser.interfaces.ITerms)
 class MappedTerms(object):
-    """A terms implementation that knows how to handle a source that was 
+    """A terms implementation that knows how to handle a source that was
     created through a source factory.
     """
 
     zope.component.adapts(zc.sourcefactory.mapping.ValueMappingSource,
-        zope.publisher.interfaces.browser.IBrowserRequest)
+                          zope.publisher.interfaces.browser.IBrowserRequest)
```

Comment:
That suprises me actually, I thought `zope.component.adapts` was not possible with Python 3. I guess we have untested code here. There should not be the claim of Python 3 support unless this package is properly ported.

Reply:
According to https://github.com/zopefoundation/zope.component/blob/e7fe69a371afd52781a3c49c130768b607bcff48/src/zope/component/_declaration.py#L39-L50 it should still work.

And yes, `src/zc/sourcefactory/browser/mapping.py` is mostly untested but this statement is executed at import time and does not raise an error.

Reply:
But it should be refactored, you are right.